### PR TITLE
chore(publish): publish packages on release

### PR DESCRIPTION
**Pull request by rryter**
- **What kind of change does this PR introduce?**
This is helping us releasing the library in a consistent manner. It will not release automatically, it will release when a github release is created.

- **What is the current behavior?**
A person manually runs the npm publish command.

- **What is the new behavior?**
To publish the package, one now can simply create a release on github, and the package will be published automatically.

### CODE REVIEW

File: .github/workflows/github-publish.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12.x'
+          registry-url: 'https://npm.pkg.github.com'
+          scope: '@erc725'
```
Comment (Hugoo) [inline .github/workflows/github-publish.yml:17]: Maybe you can enable cache? https://github.com/actions/setup-node

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12.x'
+          registry-url: 'https://registry.npmjs.org'
+          scope: '@erc725'
```
Comment (Hugoo) [inline .github/workflows/npm-publish.yml:14]: Same here with cache https://github.com/actions/setup-node

**Review by Hugoo — APPROVED** [review]
These 2 jobs look very similar.

Maybe you can use a matrix strategy to set the registry URL and token?

```
name: Node.js CI
on: [push]
jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
       include:
         - node-version: 10.x
           site: "prod"
           datacenter: "site-a"
         - node-version: 12.x
           site: "dev"
           datacenter: "site-b"
    steps:
      - name: Echo site details
        env:
          SITE: ${{ matrix.site }}
          DATACENTER: ${{ matrix.datacenter }}
        run: echo $SITE $DATACENTER

```

And do you want to use npm install or npm ci ?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12.x'
+          registry-url: ${{ matrix.registry-url }}
+          scope: '@erc725'
+          cache: 'npm'
+      - run: npm ci
+      - run: npm run build
+      - run: ${{ matrix.token }}
```
Comment (rryter) [inline .github/workflows/publish.yml]: @Hugoo do you know if this works?

**Pull request merged**